scripts/ci-build.sh

```diff
@@ -0,0 +1,512 @@
+    flag=
+    if [[ -e "$f" ]] ; then
+      if [[ $cmd == cp && -d "$f" ]] ; then
+        flag=-a
+      elif [[ $cmd == cp ]] ; then
+        flag=--preserve=all
+      fi
+      $cmd $flag "$f" staging/
+    else
+      echo "Skipping non-existing (or already moved) artifact: $f"
+    fi
+  done
+}
+
+# Layer version overrides:
+# Find and evaluate environment variables of type
+
+# LAYER_<layername>_FORK
+# LAYER_<layername>_COMMIT
+# LAYER_<layername>_BRANCH
+# LAYER_<layername>_TAG
+
+# *NOTE*  Variables cannot have dashes in name therefore, underscores must be
+# used!  For example: LAYER_meta_openembedded for meta-openembedded layer
+
+# If an override variable is defined, build with this version of the layer
+# instead.  If multiple conflicting values, then the most specific override
+# (commit) wins
+
+layer_override() {
+  local name=$1 suffix var_name value d
+
+  # Variables cannot have dashes in name therefore, underscores must be
+  # used.  For example: LAYER_meta_browser_COMMIT for meta-browser layer
+  clean_name=$(echo "$name" | sed 's/-/_/g')
+
+  # First FORK...
+  var_name="LAYER_${clean_name}_FORK"
+  value=$(deref "$var_name")
+  if [ -n "$value" ] ; then
+    echo "***** NOTE: OVERRIDING LAYER $name USING \$${var_name} = $value *****"
+    d="$PWD"
+    cd $name || echo "Layer name wrong?  Can't cd to dir : $name"
+    echo "Fetching from $value"
+    git remote add temp_fork "$value"
+    git fetch temp_fork
+
+    # Master is assumed, but it is likely overridden by a COMMIT/TAG/BRANCH
+    # value below. but this could fail.
+    echo "NOTE: Assuming master branch for now (might be overridden later)"
+    echo "NOTE: This will fail if no master branch exists in $value"
+    echo "+ git reset temp_fork/master --hard"
+    git reset temp_fork/master --hard
+    cd "$d"
+  fi
+
+  # ...then the others
+  for suffix in COMMIT TAG BRANCH ; do
+    var_name="LAYER_${clean_name}_${suffix}"
+    value=$(deref "$var_name")
+    if [ -n "$value" ] ; then
+      echo "***** NOTE: OVERRIDING LAYER $name USING \$${var_name} = $value *****"
+      d="$PWD"
+      cd $name || echo "Layer name wrong?  Can't cd to dir : $name"
+      git fetch --all
+
+      # If forked, make sure to use the right remote, otherwise origin
+      # For branches, prefer the explicit: remote/branchname
+      if git remote | grep -q temp_fork ; then
+        git checkout "temp_fork/$value" 2>/dev/null || git checkout $value
+      else
+        git checkout "origin/$value" 2>/dev/null || git checkout $value
+      fi
+
+      cd "$d"
+      break  # First one wins, priority order: COMMIT >= TAG >= BRANCH
+    fi
+  done
+}
+
+# Because of the order they are called we can't use output from the
+# layer_override function above, so we have some repeated code here.
+print_layer_overrides() {
+  local names="$1" name clean_name suffix var_name value d
+
+  for name in $names ; do
+    clean_name=$(echo "$name" | sed 's/-/_/g')
```

Comment:
Could more succinctly use string substitution directly in bash, it's just that I'm familiar with the sed syntax...